src/platform/macos/events_loop.rs

```diff
@@ -544,6 +544,67 @@ impl Proxy {
     }
 }
 
+pub fn windows_key_code_for_char_code(c: char) -> Option<c_ushort> {
+    // We only translate keys that are affected by keyboard layout.
+    //
+    // Note that since keys are translated in a somewhat "dumb" way (reading character)
+    // there is a concern that some combination, i.e. Cmd+char, causes the wrong
+    // letter to be received, and so we receive the wrong key.
+    //
+    // Codes taken from to_virtual_key_code.
+    // It may be better to map these codes to something more immediately meaningful.
```

Comment:
Is there any reason not to just have this function directly return `Option<VirtualKeyCode>`? Remapping the scancodes is a bit unintuitive.

Reply:
The original thinking was so that we can more easily "fall through" to `to_virtual_key_code()` -- the last stop to get a keycode, which must be passed through.  I realise now that it may be better (and more concise) to simply do all checking inside `to_translated_macos_scancode()` (which would then be renamed) and return a virtual keycode from there.